app/controllers/actions/PlaybackAction.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.actions
+
+import com.google.inject.{ImplementedBy, Inject}
+import connector.EnrolmentStoreConnector
+import models.requests.DataRequest
+import pages.WhatIsTheUTRVariationPage
+import play.api.Logger
+import play.api.mvc.Results.Redirect
+import play.api.mvc._
+import services.AuthenticationService
+import uk.gov.hmrc.auth.core.{BusinessKey, FailedRelationship, Relationship}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.HeaderCarrierConverter
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class PlaybackActionImpl @Inject()(val parser: BodyParsers.Default,
```

Comment:
@darrylmurphy This won't work for Agents as they're not going to have a Relationship established in IV.
Organisation users have a relationship.
For agents, we need to do the check we've built to ensure they have the correct HMRC-TERS-ORG enrolment